File: TREE/Python/LC 222 - COUNT COMPLETE TREE NODES.py
# LEETCODE 222 - Count Complete Tree Nodes
# QUESTION LINK - https://leetcode.com/problems/count-complete-tree-nodes/

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def count(self, root, ans):
        if not root:
            return
        ans[0] += 1
        self.count(root.left, ans)
        self.count(root.right, ans)

    def countNodes(self, root):
        ans = [0]  # Using list to simulate pass-by-reference
        self.count(root, ans)
        return ans[0]

File: TREE/Java/LC 222 - COUNT COMPLETE TREE NODES.java
// LEETCODE 222 - Count Complete Tree Nodes
// QUESTION LINK - https://leetcode.com/problems/count-complete-tree-nodes/

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

class Solution {
    private void count(TreeNode root, int[] ans) {
        if (root == null) return;
        ans[0]++;
        count(root.left, ans);
        count(root.right, ans);
    }

    public int countNodes(TreeNode root) {
        int[] ans = new int[1];  // Using array to simulate pass-by-reference
        count(root, ans);
        return ans[0];
    }
}
